### PR TITLE
fix strideing RQ perforamnce

### PR DESCRIPTION
**Pull request by rafiw**
Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1911/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1348/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1350/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1913/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1918/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1355/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1358/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1921/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1943/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1381/ for details (Mellanox internal link).

File: src/vma/dev/allocator.h [hunk 1]
```diff
@@ -39,28 +39,20 @@
...
-		return lkey;
-	}
+	uint32_t find_lkey_by_ib_ctx(ib_ctx_handler *p_ib_ctx_h) const;
 private:
 	bool register_memory(size_t size, ib_ctx_handler *p_ib_ctx_h, uint64_t access);
 	bool hugetlb_alloc(size_t sz_bytes);
 	// List of memory regions
-	mr_map m_mrs;
+	ibv_mr** mr_list;
+	size_t mr_list_len;
```
Comment (ilansmith) [inline src/vma/dev/allocator.h]: `m_mr_list` (or whatever the convention is for a list of pointers) and `m_mr_list_len`.

File: src/vma/dev/cq_mgr_mp.cpp [hunk 2]
```diff
@@ -30,11 +30,9 @@
  * SOFTWARE.
  */
 
-#include "dev/cq_mgr_mp.h"
-#include "dev/cq_mgr_mlx5.inl"
+#include <dev/cq_mgr_mp.h>
```
Comment (ilansmith) [inline src/vma/dev/cq_mgr_mp.cpp]: Please change to `"dev/cq_mgr_mp.h"`

File: src/vma/dev/cq_mgr_mp.cpp [hunk 3]
```diff
@@ -132,6 +129,8 @@ int cq_mgr_mp::poll_mp_cq(uint16_t &size, uint32_t &strides_used,
 			out_cqe64 = cqe;
 			size = stride_byte_cnt & MP_RQ_BYTE_CNT_FIELD_MASK;
 		}
+		// will be optimized
```
Comment (ilansmith) [inline src/vma/dev/cq_mgr_mp.cpp]: Please remove this comment.

File: src/vma/dev/net_device_val.h [hunk 4]
```diff
@@ -56,20 +56,20 @@ class L2_address;
...
 	ring_alloc_logic_attr(const ring_alloc_logic_attr &other);
-
-	/* ring allocation logic , per thread per fd ... */
-	ring_logic_t		m_ring_alloc_logic;
-	/* key in g_p_ring_profile */
-	vma_ring_profile_key	m_ring_profile_key;
-	/* either user_idx or key as defined in ring_logic_t */
-	uint64_t		m_user_id_key;
-	std::string		m_str;
+	void init();
```
Comment (ilansmith) [inline src/vma/dev/net_device_val.h]: `init()` should be private.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1949/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1950/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1387/ for details (Mellanox internal link).

File: src/vma/dev/allocator.h (hunk 1 above)
Comment (ilansmith) [inline src/vma/dev/allocator.h]: :+1:

File: src/vma/dev/cq_mgr_mp.cpp (hunk 2 above)
Comment (ilansmith) [inline src/vma/dev/cq_mgr_mp.cpp]: :+1:

File: src/vma/dev/cq_mgr_mp.cpp (hunk 3 above)
Comment (ilansmith) [inline src/vma/dev/cq_mgr_mp.cpp]: :+1:

File: src/vma/dev/net_device_val.h (hunk 4 above)
Comment (ilansmith) [inline src/vma/dev/net_device_val.h]: :+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1952/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1389/ for details (Mellanox internal link).

File: src/vma/sock/sockinfo.cpp [hunk 5]
```diff
@@ -1201,3 +1204,45 @@ int sockinfo::fast_nonblocking_rx(vma_packets_t *vma_pkts)
...
+
+	if (m_rings_fds) {
+		res_length = count;
+		return m_rings_fds;
+	}
+	// count number of fds
+	rx_ring_map_t::iterator it = m_rx_ring_map.begin();
+	for (; it != m_rx_ring_map.end(); ++it) {
+		for (int j = 0; j < it->first->get_num_resources(); ++j) {
+			++count;
```
Comment (ilansmith) [inline src/vma/sock/sockinfo.cpp]: Please do `count += it->first->get_num_resources();`

File: src/vma/sock/sockinfo.cpp [hunk 6]
```diff
@@ -1201,3 +1204,45 @@ int sockinfo::fast_nonblocking_rx(vma_packets_t *vma_pkts)
...
+
+#ifdef DEFINED_VMAPOLL
+int* sockinfo::get_rings_fds(int &res_length)
+{
+	int* channel_fds = m_p_rx_ring->get_rx_channel_fds();
+	res_length = 1;
+	return channel_fds;
+}
+#else
+int* sockinfo::get_rings_fds(int &res_length)
```
Comment (ilansmith) [inline src/vma/sock/sockinfo.cpp]: Please do the `ifdef` inside a single function of `get_rings_fds()`

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1394/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1397/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
Merged #410.

**Pull request closed without merging**